packages/@aws-cdk/cdk/lib/core/construct.ts

```diff
@@ -2,22 +2,18 @@ import cxapi = require('@aws-cdk/cx-api');
 import { makeUniqueId } from '../util/uniqueid';
 export const PATH_SEP = '/';
 
-/**
- * Represents the building block of the construct graph.
- * When a construct is created, it is always added as a child
- */
-export class Construct {
+export class Node {
```

Comment:
Can't help but feel that `Node` is overly generic. `ConstructNode` maybe, or even `ConstructTree`?

Other than that, muy bien!

Reply:
I would support both `treeNode` and `node`.